src/multivariates.jl

```diff
@@ -113,9 +113,17 @@ end
 
 ## log likelihood
 
+function _loglikelihood(d::MultivariateDistribution, X::AbstractMatrix)
+    ll = 0.0
+    @inbounds for i in 1:size(X, 2)
+        ll += _logpdf(d, view(X, :, i))
+    end
+    return ll
+end
+
 function loglikelihood(d::MultivariateDistribution, X::AbstractMatrix)
     size(X, 1) == length(d) || throw(DimensionMismatch("Inconsistent array dimensions."))
-    return sum(x -> _logpdf(d, x), X, 2)
+    return _loglikelihood(d, X)
```

Comment:
> I tend to avoid generators where possible since the eltype can't be inferred

What do you mean? I wasn't aware generators introduced any inference issues (besides the usual closure boxing issues, but this shouldn't be affected by that). AFAICT, this statement seems well-inferred to me:

```julia
julia> f(d, X) = sum(Distributions._logpdf(d, view(X, :, i)) for i in 1:size(X, 2))
f (generic function with 1 method)

julia> d = MvNormal(rand(3), rand())

julia> X = rand(g, 10);

julia> @code_warntype f(d, X)
Variables:
  #self#::#f
  d::Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}}
  X::Array{Float64,2}
  #5::##5#6{Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}},Array{Float64,2}}

Body:
  begin
      #5::##5#6{Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}},Array{Float64,2}} = $(Expr(:new, ##5#6{Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}},Array{Float64,2}}, :(d), :(X)))
      SSAValue(3) = (Base.arraysize)(X::Array{Float64,2}, 2)::Int64
      SSAValue(1) = $(Expr(:new, UnitRange{Int64}, 1, :((Base.select_value)((Base.sle_int)(1, SSAValue(3))::Bool, SSAValue(3), (Base.sub_int)(1, 1)::Int64)::Int64)))
      SSAValue(2) = $(Expr(:new, Base.Generator{UnitRange{Int64},##5#6{Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}},Array{Float64,2}}}, :(#5), SSAValue(1)))
      return $(Expr(:invoke, MethodInstance for mapfoldl(::Base.#identity, ::Function, ::Base.Generator{UnitRange{Int64},##5#6{Distributions.MvNormal{Float64,PDMats.ScalMat{Float64},Array{Float64,1}},Array{Float64,2}}}), :(Base.mapfoldl), :(Base.identity), :(Base.+), SSAValue(2)))
  end::Float64
```